chrome/content/quicktext.js

```diff
@@ -46,14 +46,13 @@ var quicktext = {
       var contentFrame = GetCurrentEditorElement();
       contentFrame.addEventListener("keypress", function(e) { quicktext.editorKeyPress(e); }, false);
 
-    	// Add an eventlistener for the popup-menu.
-    	var menu = document.getElementById("msgComposeContext");
-	    menu.addEventListener("popupshowing", function(e) { quicktext.popupshowing(e); }, false);
-
-      // Need to update GUI when the Quicktext-button is added to the toolbar.
-      var composeToolbar = (document.getElementById("composeToolbar")) ? document.getElementById("composeToolbar") : document.getElementById("composeToolbar2");
-      if (composeToolbar)
-        composeToolbar.addEventListener("DOMNodeInserted", function(e) { quicktext.toolbarButtonAdded(e); }, false);
+      // Add an eventlistener for the popup-menu.
+      var menu = document.getElementById("msgComposeContext");
+      menu.addEventListener("popupshowing", function(e) { quicktext.popupshowing(e); }, false);
```

Comment:
Do you know why anonymous functions (that don't seem to actually do anything besides calling a `quicktext.<...>` function) are used here, instead of directly specifying the `quicktext.<...>` functions? E.g. `menu.addEventListener("popupshowing", quicktext.popupshowing, false);`.

I think because anonymous function are used, all the `window.removeEventListener` in `unload` actually don't do anything. The 3 argument (`useCapture`) also need to be same for add an remove (this is not the case for the _keypress_, _keydown_ and _keyup_ listener).
And for the new _aftercustomization_ listener you forgot to add the `removeEventListener` in `unload`

Reply:
For aftercustomization : All DOM nodes with this listener attached get removed, before they are added again. Is it still needed to remove the listener?

For the anonymous function: Isn't it a rule, to always use anonymous functions? Would you suggest to just write

menu.addEventListener("popupshowing", quicktext.popupshowing, false);

Reply:
Ok, I tried, does not work. For some reason, the eventlisteners loose the quicktext context. I get this.XY is not defined (that call being part of the function assigned to the listener).

I will keep this on my mind (actually I will add an issue), for TB61 I have to rewrite many parts anyhow.

Reply:
Will this work with removeEventListener? lieser suggested to go away from attaching anonymous function, because removeEventListener only works if a real function was attached...

Reply:
I spend way to much time on this. [Here](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) you find

**A problem in the example above is that you cannot remove the listener with bind().
Another solution is using a special function called handleEvent() to catch any events**

I tried and its true. The bind stuff fixes the error, but the goal of this rewrite is not reached.

I do not want to go for handleEvent() at this stage.

Reply:
Sorry for the late reply. If there is no easy fix, I'm completely fine with leaving it as is.